### PR TITLE
Scheme

### PR DESCRIPTION
**Pull request by ralfstuckert**
Added the QRCode schemes Wifi and Girocode in addition to the already existing VCard, and moved them to new subpackage scheme.

Also added an extendable parser that creates objects like URL, Wifi, VCard etc. from any given string, but I'm not quite sure, if this is really useful. Sometimes we develop stuff just because we can ;-)

### CODE REVIEW

File: core/src/main/java/net/glxn/qrgen/core/scheme/VCard.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	private String email;
+	private String address;
+	private String website;
+	private String note;
+
+	public VCard() {
+	}
+
+	public VCard(final String vcardCode) {
+		if (vcardCode == null || !vcardCode.startsWith(BEGIN_VCARD)) {
```
Comment (kenglxn) [inline core/src/main/java/net/glxn/qrgen/core/scheme/VCard.java]: This breaks the existing VCARD impl. Is there a need for a constructor that overrides the vcardcode. If not I suggest keeping the name string as the default to ensure backwards compat.

**Comment by kenglxn** [comment]
Looks cool, a lot of improvements here. But I can not merge until the build is passing.

There are compilation failures due to 
./QRGen/javase/src/main/java/net/glxn/qrgen/javase/QRCode.java and ./QRGen/javase/src/test/java/net/glxn/qrgen/javase/QRCodeTest.java using the old card impl. 

Instead of moving the VCARD and opening for deprecated classes I would prefer extending behaviour in the existing class. This leads to less breakage. 

I am fine with moving the class to a new package, I just don't want two implementations laying around in different packages where one is deprecated.

**Comment by ralfstuckert** [comment]
Hi Ken
Sorry for the mess. Hopefully everything is fine now.
Since I'm new to pull requests: do I have to do another pull request, or will you just try again?
Thanks for your patience
Ralf
Am 25.07.2015 22:09 schrieb Ken Gullaksen <notifications@github.com>:Looks cool, a lot of improvements here. But I can not merge until the build is passing.

There are compilation failures due to 
./QRGen/javase/src/main/java/net/glxn/qrgen/javase/QRCode.java and ./QRGen/javase/src/test/java/net/glxn/qrgen/javase/QRCodeTest.java using the old card impl. 

Instead of moving the VCARD and opening for deprecated classes I would prefer extending behaviour in the existing class. This leads to less breakage. 

I am fine with moving the class to a new package, I just don't want two implementations laying around in different packages where one is deprecated.

**Comment by kenglxn** [comment]
Great stuff. Thanks for the contribution.

**Comment by atomfrede** [comment]
:+1:

**Pull request merged**